BNote/src/presentation/widgets/filebrowser.php

```diff
@@ -280,64 +257,152 @@ private function writeFolderContent() {
 			}
 			$dirname = str_replace("/", " > ", $dirname);
 			Writing::h3($dirname, "filebrowser_folder_header");
-			
+
 			// show table with files
-			$content = $this->getFilesFromFolder($this->root . $this->path);
-			?>
-			<div class="filebrowser_filepanel">
-				<?php
-				$this->writeFolderContentItem($content, "folder");
-				$this->writeFolderContentItem($content, "file");
-				?>
-			</div>
-			<?php
+			$content = $this->getFilesFromFolder($this->path);
```

Comment:
If you change `$this->root . $this->path` to `$this->path` the root context is lost - see lines 96-98

Reply:
The root context isn't lost. The concatenation $this->root ./this->path is done inside the method getFilesFromFolder. (See changes in line 565 in the same commit.)
As far as I can see it, the folder argument to getFilesFromTolder is not really neeeded at all, because both $root and $path are members and passed through the this pointer.
Did I miss something.

Reply:
OK, I took this change back for now, because I found some problems with ill formed paths that caused strange behavior.
I still believe that moving the concatenation of $this->root .$this->path into getFilesFromFolder is not a problem.
I think I messed up the some string construction.

I'll have to do some deeper investigation. I think a double slash should not occur at all. I wonder why it works anyway...